src/api/course/index.ts

```diff
@@ -84,23 +94,137 @@ export class CourseApi {
       if (storeData) {
         this.emitCourseInfo(true)
       }
+      // register delta handler
+      ;(this.app as any).registerDeltaInputHandler(this.deltaHandler)
       resolve()
     })
   }
 
+  /** handle and process incoming course stream data */
+  private deltaHandler = (
+    delta: ValuesDelta,
+    next: (delta: ValuesDelta) => void
+  ) => {
+    this.processStreamSources(delta)
+    next(delta)
+  }
+
+  /** Process stream <destination>.nextPoint data
+   * Note: Stream source cannot override destination set by API!
+   * Destination is set when:
+   * 1. There is no current destination
+   * 2. msg source matches current Destination source
+   * 3. Destination Position is changed.
+   */
+  private async processStreamSources(delta: ValuesDelta) {
+    if (!this.cmdSource || (this.cmdSource && this.cmdSource.type !== 'API')) {
+      delta.updates.forEach((update: Update) => {
+        if (
+          !update.values ||
+          (update.values && !Array.isArray(update.values))
+        ) {
+          return
+        }
+        update.values.forEach((pathValue: PathValue) => {
+          if (
+            pathValue.path ===
+              'navigation.courseRhumbline.nextPoint.position' ||
+            pathValue.path === 'navigation.courseGreatCircle.nextPoint.position'
+          ) {
+            if (
+              update.source &&
+              update.source.type &&
+              ['NMEA0183', 'NMEA2000'].includes(update.source.type)
+            ) {
+              this.parseStreamValue(
+                {
+                  type: update.source.type,
+                  label: update.source.label,
+                  msg:
+                    update.source.type === 'NMEA0183'
+                      ? `${update.source.sentence}:${update.source.talker}`
```

Comment:
Instead of rolling out your own ids I think this should use `getSourceId`. Specifically in N2K `src` is error prone, because the bus may shuffle addresses.

Reply:
Updated to use getSourceId.